File: server/AniStream.Models/Models/ProfileModel.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using Microsoft.EntityFrameworkCore;

namespace AniStream.Models;

[PrimaryKey(nameof(ProfileId))]
[Table("profile")]
public sealed class ProfileModel
{
    public int ProfileId { get; set; }

    public string Uuid { get; set; }

    public string Name { get; set; }

    public string Password { get; set; }

    public string PasswordSalt { get; set; }
    
    public string BackgroundColor { get; set; }

    public string Eye { get; set; }

    public string Mouth { get; set; }

    public string Theme { get; set; }

    public string Lang { get; set; }

    public bool TosAccepted { get; set; }

    public bool SyncCatalog { get; set; }

    public ProfileModel(
        int profileId,
        string uuid,
        string name,
        string password,
        string passwordSalt,
        string backgroundColor,
        string eye,
        string mouth,
        string theme,
        string lang,
        bool tosAccepted,
        bool syncCatalog
    )
    {
        ProfileId = profileId;
        Uuid = uuid;
        Name = name;
        Password = password;
        PasswordSalt = passwordSalt;
        BackgroundColor = backgroundColor;
        Eye = eye;
        Mouth = mouth;
        Theme = theme;
        Lang = lang;
        TosAccepted = tosAccepted;
        SyncCatalog = syncCatalog;
    }

    public ProfileModel(
        string uuid,
        string name,
        string password,
        string passwordSalt,
        string backgroundColor,
        string eye,
        string mouth,
        string theme,
        string lang,
        bool tosAccepted,
        bool syncCatalog
    ) : this(
        0,
        uuid,
        name,
        password,
        passwordSalt,
        backgroundColor,
        eye,
        mouth,
        theme,
        lang,
        tosAccepted,
        syncCatalog
    )
    {
    }
}

File: dashboard/src/models/profile.model.ts
import type {botttsNeutral} from "@dicebear/collection";

export type ProfileEye = NonNullable<botttsNeutral.Options["eyes"]>[0];
export type ProfileMouth = NonNullable<botttsNeutral.Options["mouth"]>[0];

export type SupportedLocals = "de" | "en";

export interface ProfileModel {
    profile_id: number;
    uuid: string;
    name: string;
    background_color: string;
    eye: ProfileEye;
    mouth: ProfileMouth;
    theme: string;
    lang: SupportedLocals;
    tos_accepted: boolean;
    dashboard_user: boolean;
    client_user: boolean;
}

export interface ProfileUpdateModel {
    name: string;
    background_color: string;
    eye: ProfileEye;
    mouth: ProfileMouth;
    theme: string;
    lang: "de" | "en";
    tos_accepted: boolean;
    dashboard_user: boolean;
    client_user: boolean;
}


export interface ProfileCreateModel {
    name: string;
    background_color: string;
    eye: ProfileEye;
    mouth: ProfileMouth;
    theme: string;
    lang: "de" | "en";
    password: string;
    dashboard_user: boolean;
    client_user: boolean;
}